test/unit/resources/mock/iam/aws_sns_topic_mock.rb

```diff
@@ -0,0 +1,26 @@
+require_relative '../aws_base_resource_mock'
+
+class AwsSnsTopicMock < AwsBaseResourceMock
+
+  def initialize
+    super
+    @topic = {}
+    @attributes = {"TopicArn"=> @aws.any_arn,
+                   "KmsMasterKeyId"=> "alias/aws/sns"}
+    @topic[:attributes] = @attributes
+  end
+
+  def stub_data
+    stub_data = []
+
+    topic = {:client => Aws::SNS::Client,
+             :method => :get_topic_attributes,
+             :data => @topic}
+
+    stub_data += [topic]
```

Comment:
why not simply?

```suggestion
    topic = {
      client: Aws::SNS::Client,
      method: :get_topic_attributes,
      data: @topic
    }
    [topic]
```

Reply:
Since for rest of the resources, this method was used. I followed similar procedure.

Reply:
Okay, If this makes better sense, could you update it..?  Old resources may need some refactoring in the future..

Reply:
Definitely, I will make that commit in a minute.

Reply:
you could commit the suggestion directly if you would like to ..

Reply:
committed necessary changes